design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,169 @@
+# Adding and removing Ceph osds via Ceph orchestrator interface
+
+## Goals
+
+1. Give Ceph's Rook Orchestrator Plugin the ability to add and remove OSDs to/from a Rook-Ceph
+   cluster running in Kubernetes.
+2. Users of the Ceph Orchestrator CLI and/or Ceph Dashboard should be able to
```

Comment:
Is the goal to expose all capabilities of ceph-volume I see this as feasible. ceph-volume doesn't even expose all the ways possible to create OSDs with Ceph. I would also argue we don't want to expose all the c-v options. For example, why expose filestore? Its lifetime is very limited at this point.

Reply:
WIP around OSD creation in the Ceph orchestrators group: The use of drive groups and the translation of the drive groups  to C-V commands is being discussed in the [mgr/cephadm: extend support for drivegroups](https://github.com/ceph/ceph/pull/32545/).
I think it follows the main goals defined here. The orchestrator is responsable to define the right command/set of commands that are executed in k8s using jobs. Only one concern,  the management of the cluster based in CRD's is going to be "damaged" if we do not keep a synchronization between CRD's and the OSD's added/removed using the orchestrator.

Reply:
I believe we're in agreement since the OSDs are now only being added via the CR